### PR TITLE
implementing numpy.round frontend function and tests.

### LINKED ISSUE #19015: round

### PR DESCRIPTION
**Pull request by salamonyfares**
Close #19015 

I am facing a problem while running the specific test for my implemented function.
whenever I run the test, it throws a key error with the value "cpu" and I have traced back the error and I have found that it is trying to read a key called "cpu" in the function called "_get_fn_dtypes" and the key should be corresponding to this variable "test_globals.CURRENT_DEVICE_STRIPPED"; but when I print out the values of this dictionary, I only find one key which is the backend name and no cpu key. I believe that this is something related to my local configurations, if yes please let me know what should I do to overcome this problem and see if there are any other problems in the test.

### CODE REVIEW

**Comment by salamonyfares** [comment]
Close #19015 
the error in the above comment got solved once I pulled the latest change on the master branch, I ran the tests and they ran successfully as the YouTube video on implementing numpy frontend function.

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Hey @salamonyfares thanks for making the pull request! This looks good, the only thing is numpy.round_ is deprecated as of numpy 1.25.0 https://numpy.org/devdocs/release/1.25.0-notes.html. Could we change this frontend function and test to be numpy.round instead?

**Comment by salamonyfares** [comment]
Hello @Sam-Armstrong, I have made the requested changes and pulled the latest master branch into my fork to make my pull request as up-to-date as possible. Also, I have changed the name of the function in my issue #19015 to make it the same as you requested as well. kindly, have a look and let me know if there are any changes. Kind Regards.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py [hunk 1]
```diff
@@ -240,3 +240,32 @@ def test_numpy_around(
...
         decimals=decimals,
     )
+
+
+# round
+@handle_frontend_test(
+    fn_tree="numpy.round",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+    ),
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py:250]:
```suggestion
        available_dtypes=helpers.get_dtypes("float"),
        max_value=100.,  # to stop overflows with float16
        min_value=-100.,
    ),
```

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py [hunk 2]
```diff
@@ -240,3 +240,32 @@ def test_numpy_around(
...
+    decimals=st.integers(min_value=0, max_value=5),
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py:251]:
```suggestion
    decimals=st.integers(min_value=0, max_value=3),
```

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
Looks like the tests are failing right now because of float16 overflow inconsistencies. I think we can make changes like these just to get the tests passing. You can run the tests with pytest ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_rounding.py::test_numpy_round to check they're working

**Comment by salamonyfares** [comment]
I have tried the suggested fix you mentioned but I still faced the same issue as sometimes all the tests succeed and sometimes 2 out of the 5 tests fail, the 2 tests failing were in the JAX and paddle backends, and I have noticed from multiple runs that they fail whenever the test case is a number smaller than "-60." so I have sat the minimum to "-60." and ran the tests multiple times and they succeeded. Hope this makes us able to close the issue now. waiting for your feedback

**Review by Sam-Armstrong — CHANGES_REQUESTED** [review]
@salamonyfares I actually this is a problem with the ivy functional api/backend implementations of round(). We are better to address these than modify our test, so I'll ask the framework team about this. In the meantime, please could you remove the changes you made to .idea/runConfigurations/_template__of_py_test.xml, and change the test back to how it was before (no min/max value for dtype_and_x, decimals max_value=5, etc)? Thanks! 😊

**Comment by salamonyfares** [comment]
Hello @Sam-Armstrong, I have done the changes your requested. But as I was searching the problem I came across something that might be helpful in our situation which is I found that this is an error in the original numpy library which is something related to the maximum number of float16. I am going to attach the links I found so you can check them as well. Let me know if there is anything that is required from me. Regards 😊
**the links:**
- https://stackoverflow.com/questions/56368092/why-does-python-roundnp-float16np-pi-5-return-infinity-bug-limitation-or
- https://github.com/numpy/numpy/issues/13699
- https://github.com/numpy/numpy/issues/21941

**Comment by salamonyfares** [comment]
Hey @Sam-Armstrong, I am very sorry for what I did there in the last couple of commits; as I was trying to revert back the commit to a commit before modifying the .idea folder but I faced some issues that caused changes in the code in other places so I ended up deleting all the commits up until before you suggest the limits on the float16 numbers. and that made my fork out of date so I updated my fork using the original repo's master branch to have all the commits since then (you will find that in the commit with the id "0ae71a3") and then made the changes that were deleted in my work so now we are at a point that all the changes required are done without any change in the .idea folder nor any change in any code that is not mine. However this accident made me ask for a review from the people whom I have their code changed by mistake in the wrong revert I did, but this is not the case anymore which means that there is no change in anyone's code so I am very sorry @KareemMAX , @CatB1t , @vedpatwardhan for the inconvenience and the false notification.  😊
Kindly @Sam-Armstrong, check all the commits and let me know if there is still anything you find wrong and let me know so I can update it or make any necessary changes. Again, sorry for all that and I hope it is ok 🙏🏻.

**Review by Sam-Armstrong — APPROVED** [review]
@salamonyfares No worries! I talked to the framework team and they said it's a limitation of the ivy framework right now, so I reinstated the changes we made to exclude any overflow cases from the test. Thanks for your work on this PR, we can now merge! 😊

**Comment by salamonyfares** [comment]
@Sam-Armstrong, that's great!
Thank you very much for the support Sam, I appreciate it! 😊

**Pull request merged**